analysers/trials/constructDuration.py

```diff
@@ -0,0 +1,97 @@
+import sys
+import json
+import io
+import gzip
+import uuid
+import math
+
+from datetime import timedelta
+
+import numpy as np
+
+from pyspark_cassandra import CassandraSparkContext
+from pyspark_cassandra import RowFormat
+from pyspark import SparkConf
+
+def createQuery(sc, dataRDD, experimentID, trialID):
+    from commons import computeMode, computeMetrics
+    
+    queries = []
+    
+    modeRDD = dataRDD.filter(lambda r: r['duration'] != None) \
+            .map(lambda r: (r['duration'], 1)) \
+            .cache()
+                   
+    mode = computeMode(modeRDD)
+    
+    data = dataRDD.map(lambda r: r['duration']).collect()
+    metrics = computeMetrics(data)
+    
+    queries.append({"construct_name":"all", "construct_type":"all", "experiment_id":experimentID, "trial_id":trialID, "construct_duration_mode":mode[0], \
+                    "construct_duration_mode_freq":mode[1], "construct_duration_p90":metrics["p90"], "construct_duration_p99":metrics["p99"], \
+                    "construct_duration_percentiles":metrics["percentiles"], \
+                  "construct_duration_mean":metrics["mean"], "construct_duration_num_data_points":metrics["num_data_points"], \
+                  "construct_duration_min":metrics["min"], "construct_duration_max":metrics["max"], "construct_duration_sd":metrics["sd"], \
+                  "construct_duration_q1":metrics["q1"], "construct_duration_q2":metrics["q2"], "construct_duration_q3":metrics["q3"], "construct_duration_p95":metrics["p95"], \
+                  "construct_duration_me":metrics["me"], "construct_duration_ci095_min":metrics["ci095_min"], "construct_duration_ci095_max":metrics["ci095_max"]})
+    
+    constructs = dataRDD.map(lambda a: a["construct_name"]).distinct().collect()
```

Comment:
@VincenzoFerme The reason why I check afterwards is because I can't save None types to the trial table, as those are part of the Primary Key. I do however want to select processes with type or name as None in the computations either way.

Leaving a comment in the code about this